modules/database/funcs.go

```diff
@@ -229,3 +232,36 @@ func (d *Database) getDBWithName(
 
 	return nil
 }
+
+//
+// Returns a *Database object with specified name and namespace
+//
+func GetDatabaseByName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+) (*Database, error) {
+	// create a Database by suppplying a resource name
+	db := &Database{
+		databaseName: name,
+	}
+	// then querying the MariaDBDatabase and store it in db by calling
+	if err := db.getDBWithName(ctx, h); err != nil {
+		return db, err
+	}
+	return db, nil
+}
+
+//
+// Delete a finalizer by its object
```

Comment:
comment needs to start with DeleteFinalizer to make lint happy

Reply:
done